datafusion/datasource/src/write/demux.rs

```diff
@@ -329,7 +322,7 @@ async fn hive_style_partitions_demuxer(
 
             // Finally send the partial batch partitioned by distinct value!
             part_tx.send(final_batch_to_send).await.map_err(|_| {
-                DataFusionError::Internal("Unexpected error sending parted batch!".into())
+                exec_datafusion_err!("Unexpected error sending parted batch!")
```

Comment:
I think this should be an internal error
```suggestion
                internal_datafusion_err!("Unexpected error sending parted batch!")
```